src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs

```diff
@@ -157,7 +154,11 @@ int IndexOfRest(byte value, int firstLength)
         public ReadOnlyBytes Slice(Cursor from)
         {
             if (from._node == null) return Slice(First.Length - from._index);
-            return new ReadOnlyBytes(from._node).Slice(from._index);
+            var headIndex = _all.Index + _all.First.Length - _first.Length;
```

Comment:
Could this result in integer overflow?

long + int - int => overflow

Reply:
you mean if index is approaching long.MaxValue?

Reply:
Yes.

Reply:
I am pretty sure we would hit OOM before we allocate 4 billion linked list nodes :-)